modulemd/v2/modulemd-translation-entry.c

```diff
@@ -0,0 +1,623 @@
+    }
+}
+
+
+static void
+modulemd_translation_entry_set_property (GObject *object,
+                                         guint prop_id,
+                                         const GValue *value,
+                                         GParamSpec *pspec)
+{
+  ModulemdTranslationEntry *self = MODULEMD_TRANSLATION_ENTRY (object);
+
+  switch (prop_id)
+    {
+    case PROP_LOCALE:
+      modulemd_translation_entry_set_locale (self, g_value_get_string (value));
+      break;
+
+    case PROP_SUMMARY:
+      modulemd_translation_entry_set_summary (self,
+                                              g_value_get_string (value));
+      break;
+
+    case PROP_DESCRIPTION:
+      modulemd_translation_entry_set_description (self,
+                                                  g_value_get_string (value));
+      break;
+
+    default: G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+
+static void
+modulemd_translation_entry_class_init (ModulemdTranslationEntryClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->finalize = modulemd_translation_entry_finalize;
+  object_class->get_property = modulemd_translation_entry_get_property;
+  object_class->set_property = modulemd_translation_entry_set_property;
+
+  properties[PROP_LOCALE] = g_param_spec_string (
+    "locale",
+    "Locale",
+    "The locale for this translation entry. It must correspond to the format "
+    "specified by libc locale names. This field may only be set on object "
+    "construction and is immutable afterwards.",
+    TE_DEFAULT_STRING,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT_ONLY);
+
+  properties[PROP_SUMMARY] =
+    g_param_spec_string ("summary",
+                         "Summary",
+                         "The summary of this module stream translated into "
+                         "the language specified by locale.",
+                         TE_DEFAULT_STRING,
+                         G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+
+  properties[PROP_DESCRIPTION] =
+    g_param_spec_string ("description",
+                         "Description",
+                         "The description of this module stream translated "
+                         "into the language specified by locale.",
+                         TE_DEFAULT_STRING,
+                         G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+
+  g_object_class_install_properties (object_class, N_PROPS, properties);
+}
+
+
+static void
+modulemd_translation_entry_init (ModulemdTranslationEntry *self)
+{
+  self->profile_descriptions =
+    g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+}
+
+
+/* === YAML Functions === */
+
+static void *
+modulemd_translation_entry_parse_yaml_profiles (yaml_parser_t *parser,
+                                                ModulemdTranslationEntry *te,
+                                                GError **error)
+{
+  MMD_INIT_YAML_EVENT (event);
```

Comment:
Add `MMD_INIT_TRACE()` here as well. It's intended for tracing the parsing and emitting functions and all their nested calls.